connect-inject/container_init.go

```diff
@@ -286,29 +295,40 @@ name = "{{ .ServiceName }}"
 protocol = "{{ .ServiceProtocol }}"
 EOF
 {{- end }}
+
 {{- if .AuthMethod }}
 /bin/consul login -method="{{ .AuthMethod }}" \
   -bearer-token-file="/var/run/secrets/kubernetes.io/serviceaccount/token" \
   -token-sink-file="/consul/connect-inject/acl-token" \
+  {{- if .ConsulNamespace }}
+  -ns="{{ .ConsulNamespace }}" \
+  {{- end }}
   -meta="pod=${POD_NAMESPACE}/${POD_NAME}"
 {{- /* The acl token file needs to be read by the lifecycle-sidecar which runs
        as non-root user consul-k8s. */}}
 chmod 444 /consul/connect-inject/acl-token
 {{- end }}
+
 {{- if .WriteServiceDefaults }}
 {{- /* We use -cas and -modify-index 0 so that if a service-defaults config
        already exists for this service, we don't override it */}}
 /bin/consul config write -cas -modify-index 0 \
   {{- if .AuthMethod }}
   -token-file="/consul/connect-inject/acl-token" \
   {{- end }}
+  {{- if .ConsulNamespace }}
+  -ns="{{ .ConsulNamespace }}" \
+  {{- end }}
   /consul/connect-inject/service-defaults.hcl || true
 {{- end }}
 
 /bin/consul services register \
   {{- if .AuthMethod }}
   -token-file="/consul/connect-inject/acl-token" \
   {{- end }}
+  {{- if .ConsulNamespace }}
+  -namespace={{ .ConsulNamespace }} \
```

Comment:
it's a different flag in this case? namespace/ns?

Reply:
The code currently is implemented as `-ns`, but will be switched to `-namespace` for the final release.